docs/Ingress.md

```diff
@@ -1,59 +1,876 @@
+    - "cluster-master.splunk.example.com"
+    - "license-master.splunk.example.com"
+```
+
+</p>
+</details>
+
+Next, you will need to create VirtualServices for each of the components that you want to expose outside of Kubernetes:
+
+<details><summary>
+Create one virtual service for each component
+</summary>
+<p>
+
+```yaml
+apiVersion: networking.istio.io/v1beta1
+kind: VirtualService
+metadata:
+  name: splunk
+spec:
+  hosts:
+  - "splunk.example.com"
+  gateways:
+  - "splunk-gw"
+  http:
+  - match:
+    - uri:
+        prefix: "/services/collector"
+    route:
+    - destination:
+        port:
+          number: 8088
+        host: splunk-example-indexer-service
+  - route:
+    - destination:
+        port:
+          number: 8000
+        host: splunk-example-search-head-service
+---
+apiVersion: networking.istio.io/v1beta1
+kind: VirtualService
+metadata:
+  name: splunk-deployer
+spec:
+  hosts:
+  - "deployer.splunk.example.com"
+  gateways:
+  - "splunk-gw"
+  http:
+  - route:
+    - destination:
+        port:
+          number: 8000
+        host: splunk-example-deployer-service
+---
+apiVersion: networking.istio.io/v1beta1
+kind: VirtualService
+metadata:
+  name: splunk-cluster-master
+spec:
+  hosts:
+  - "cluster-master.splunk.example.com"
+  gateways:
+  - "splunk-gw"
+  http:
+  - route:
+    - destination:
+        port:
+          number: 8000
+        host: splunk-example-cluster-master-service
+---
+apiVersion: networking.istio.io/v1beta1
+kind: VirtualService
+metadata:
+  name: splunk-license-master
+spec:
+  hosts:
+  - "license-master.splunk.example.com"
+  gateways:
+  - "splunk-gw"
+  http:
+  - route:
+    - destination:
+        port:
+          number: 8000
+        host: splunk-example-license-master-service
+```
+</p>
+</details>
+
+Finally, you will need to create a DestinationRule to ensure user sessions are
+sticky to specific search heads:
+
+<details><summary>
+DestinationRule for the Search Head
+</summary>
+<p>
+
+```yaml
+apiVersion: networking.istio.io/v1beta1
+kind: DestinationRule
+metadata:
+  name: splunk-search-head-rule
+spec:
+  host: splunk-example-search-head-service
+  trafficPolicy:
+    loadBalancer:
+      consistentHash:
+        httpCookie:
+          name: SPLUNK_ISTIO_SESSION
+          ttl: 3600s
+```
+
+</p>
+</details>
+
+
+### Configuring Ingress for Splunk Forwarder data
+The pre-requisites for enabling inbound communications from Splunk Forwarders to the cluster are configuring the Istio Gateway and Istio Virtual Service:
+
+#### Gateway
+
+The [Istio Gateway](https://istio.io/latest/docs/reference/config/networking/gateway/) defines a load balancer at the edge of the cluster that serves as the entry point for external traffic into the Kubernetes cluster. The gateway is configured with the specific port and protocol combinations that data will be sent to on the load balancer. How the traffic is routed to services is handled by the Virtual Service configuration.
```

Comment:
If we are going to leave these descriptions for gateway and virtual service, perhaps we can move them to the top of the Istio section since we already show gateway and virtual service configurations above for splunkweb access. Alternatively, maybe we can just remove these descriptions as they are istio docs.

Reply:
Done